nin/frontend/app/scripts/controllers/main.js

```diff
@@ -240,6 +242,7 @@
 
             case 'shader':
               var indirectEval = eval;
+              $scope.fileCache[event.path] = event.content;
```

Comment:
You will need to add more fileCaches to intercept the original loads as well i believe.
Layers can be loaded by the 'layers' or 'layer' event afaik, which atm contain duplicated code. the 'layers' should probably just call some function shared with the 'layer' event.